thirdeye/thirdeye-frontend/app/pods/components/rootcause-metrics/component.js

```diff
@@ -56,23 +56,23 @@ export default Ember.Component.extend({
           break;
 
         case ROOTCAUSE_METRICS_SORT_PROPERTY_DATASET:
-          output = metricUrns.map(urn => [datasets[urn], urn]).sort();
+          output = metricUrns.map(urn => [datasets[urn], metrics[urn], urn]).sort();
           break;
 
         case ROOTCAUSE_METRICS_SORT_PROPERTY_CHANGE:
-          output = metricUrns.map(urn => [makeSortable(changes[urn]), urn]).sort((a, b) => a[0] - b[0]);
+          output = metricUrns.map(urn => [makeSortable(changes[urn]), metrics[urn], urn]).sort(this._compareMixedTuple);
           break;
 
         case ROOTCAUSE_METRICS_SORT_PROPERTY_SCORE:
-          output = metricUrns.map(urn => [makeSortable(scores[urn]), urn]).sort((a, b) => a[0] - b[0]);
+          output = metricUrns.map(urn => [makeSortable(scores[urn]), metrics[urn], urn]).sort(this._compareMixedTuple);
           break;
       }
 
       if (sortMode === ROOTCAUSE_METRICS_SORT_MODE_DESC) {
         output = output.reverse();
       }
 
-      return output.map(t => t[1]);
+      return output.map(t => t[2]);
```

Comment:
Let's include some more comments for why there are 3 entries for each array mapped to a `urn` and why we're returning a mapping to the 2nd entry.

Reply:
it's a tuple sort